pkg/manager/manager.go

```diff
@@ -503,6 +509,13 @@ func (m *Manager) Recover(ctx context.Context) (map[string]*daemon.Daemon, map[s
 		var d, _ = daemon.NewDaemon(opt...)
 		d.States = *s
 
+		// It can't change snapshotter's fs driver to a different one from a daemon that ever created in the past.
+		if d.States.FsDriver != m.FsDriver {
+			return errors.Wrapf(errdefs.ErrInvalidArgument,
+				"daemon's fs driver %s mismatches nydus-snapshotter's %s",
```

Comment:
`can't recover from the last restart, the specified fs-driver=%s option mismatches with the last fs-driver=%s` may be better.

Reply:
fixed